### ISSUE TITLE
OAEP decryption is implemented ?

### ISSUE THREAD

**Issue by adelapie** [tpm2-software/tpm2-tools#2122]
My code:

```
openssl genrsa -out private.pem 2048
openssl rsa -in private.pem -out public.pem -outform PEM -pubout

echo "hello world" > plain.txt
openssl rsautl -oaep -encrypt -inkey public.pem -pubin -in plain.txt \
-out plain.rsa.enc

tpm2 loadexternal -G rsa  -r private.pem -c key.ctx
tpm2 rsadecrypt -s oaep -c key.ctx  -o plain.rsa.dec plain.rsa.enc

diff plain.txt plain.rsa.dec
```

It always fail with:
```
WARNING:esys:src/tss2-esys/api/Esys_RSA_Decrypt.c:305:Esys_RSA_Decrypt_Finish() Received TPM Error 
ERROR:esys:src/tss2-esys/api/Esys_RSA_Decrypt.c:102:Esys_RSA_Decrypt() Esys Finish ErrorCode (0x000002c3) 
ERROR: Esys_RSA_Decrypt(0x2C3) - tpm:parameter(2):hash algorithm not supported or not appropriate
```

**Comment by idesai** [comment]
@adelapie, Added a fix in the PR #2127 

Your sequence requires a slight change.

```bash
openssl genrsa -out private.pem 2048
openssl rsa -in private.pem -out public.pem -outform PEM -pubout
tpm2 loadexternal -G rsa  -r private.pem -c key.ctx

echo "hello world" > plain.txt
tpm2 rsaencrypt -c key.ctx -o plain.rsa.enc -s oaep < plain.txt

tpm2 rsadecrypt -s oaep -c key.ctx  -o plain.rsa.dec plain.rsa.enc

diff plain.txt plain.rsa.dec
```

**Comment by adelapie** [comment]
Hello @idesai , thanks for your changes ! However, after applying the changes I still get the same error: "ERROR: Esys_RSA_Decrypt(0x2C3) - tpm:parameter(2):hash algorithm not supported or not appropriate
". I've tried to change the hash algorithm with SHA1 in case SHA256 was not supported in my TPM, but with both options rsaencrypt/rsadecrypt fail.
